src/core/atomic.d

```diff
@@ -1273,6 +1182,146 @@ else version( AsmX86_64 )
     }
 }
 
+/* ***
+ * Functions built in to the compiler, along with a default implementation.
+ */
+
+bool cmpxchg(shared(ubyte)* here, ubyte ifThis, ubyte writeThis) pure nothrow @nogc @trusted
```

Comment:
@klickverbot yeah, cas is enough on my side too. I even pass the MemoryOrder value to my backend and it will do the right thing!

Having this as a publicly exposed builtin is as much use as y2lx.